packages/clay-core/src/picker/Picker.tsx

```diff
@@ -382,6 +382,20 @@ export function Picker<T extends Record<string, any> | string | number>({
 		);
 	}
 
+	const getComponentWidth = () => {
+		if (typeof width === 'string') {
```

Comment:
I would say that we could just support the use of `auto`, otherwise this could be an escape hatch to get around the imposed limits.

Reply:
Sure! Thanks @matuzalemsteles

Reply:
I don't think we need to make 'auto' a part of the api. I think this would actually just be a part of the styles we apply. To make this work I think it will need width and minWidth, like this:

```
	let menuMinWidth = '160px';
	let menuWidth = 'auto';

	if (typeof width === 'number') {
		menuMinWidth = `${triggerRef.current?.clientWidth || 0}px`;
		menuWidth = `${width}px`;
	} else if ((triggerRef.current?.clientWidth || 0) > 160) {
		menuMinWidth = `${triggerRef.current?.clientWidth}px`;
	}
```

This will result in the following situations all being possible which align with [the initial requirements](https://liferay.atlassian.net/browse/LPD-4232):

![Screenshot 2024-06-21 at 10 48 30 AM](https://github.com/liferay/clay/assets/1609196/3cf89938-03cc-4113-9585-f91f36ebdbfc)
![Screenshot 2024-06-21 at 10 51 53 AM](https://github.com/liferay/clay/assets/1609196/4c626a7a-5ca2-40ae-9488-2cf16601ce9c)
![Screenshot 2024-06-21 at 10 52 05 AM](https://github.com/liferay/clay/assets/1609196/5cbc8e37-002e-40bf-b05c-264462ebe9bb)
![Screenshot 2024-06-21 at 10 52 43 AM](https://github.com/liferay/clay/assets/1609196/0516729b-94e5-4042-bf4a-2c2148aae839)
![Screenshot 2024-06-21 at 10 53 28 AM](https://github.com/liferay/clay/assets/1609196/227971d1-c475-4e01-8cd8-27b2ca7e0c4f)

Reply:
Thanks @ethib137!
I've made the changes you requested and tested to cover all cases.